File: invision-mobile/lib/src/core/enums/stock_movement_type.dart
import 'package:flutter/material.dart';

enum StockMovementType {
  stockIn,
  stockOut,
  adjustment,
  stockReturn,
  sellOut,
  sellThrough;

  static StockMovementType fromString(String value) {
    return switch (value) {
      'stock_in' => StockMovementType.stockIn,
      'stock_out' => StockMovementType.stockOut,
      'adjustment' => StockMovementType.adjustment,
      'return' => StockMovementType.stockReturn,
      'sell_out' => StockMovementType.sellOut,
      'sell_through' => StockMovementType.sellThrough,
      _ => StockMovementType.stockIn,
    };
  }

  String get value => switch (this) {
    StockMovementType.stockIn => 'stock_in',
    StockMovementType.stockOut => 'stock_out',
    StockMovementType.adjustment => 'adjustment',
    StockMovementType.stockReturn => 'return',
    StockMovementType.sellOut => 'sell_out',
    StockMovementType.sellThrough => 'sell_through',
  };

  String get label => switch (this) {
    StockMovementType.stockIn => 'Stock In',
    StockMovementType.stockOut => 'Stock Out',
    StockMovementType.adjustment => 'Adjustment',
    StockMovementType.stockReturn => 'Return',
    StockMovementType.sellOut => 'Sell Out',
    StockMovementType.sellThrough => 'Sell Through',
  };

  Color get color => switch (this) {
    StockMovementType.stockIn => Colors.green,
    StockMovementType.stockOut => Colors.red,
    StockMovementType.adjustment => Colors.orange,
    StockMovementType.stockReturn => Colors.blue,
    StockMovementType.sellOut => Colors.indigo,
    StockMovementType.sellThrough => Colors.purple,
  };
}

File: invision-backend/app/Enums/StockMovementType.php
<?php

namespace App\Enums;

enum StockMovementType: string
{
    case StockIn = 'stock_in';
    case StockOut = 'stock_out';
    case Adjustment = 'adjustment';
    case Return = 'return';
    case SellOut = 'sell_out';
    case SellThrough = 'sell_through';

    public function label(): string
    {
        return match ($this) {
            self::StockIn => 'Stock In',
            self::StockOut => 'Stock Out',
            self::Adjustment => 'Adjustment',
            self::Return => 'Return',
            self::SellOut => 'Sell Out',
            self::SellThrough => 'Sell Through',
        };
    }

    public function color(): string
    {
        return match ($this) {
            self::StockIn => 'green',
            self::StockOut => 'red',
            self::Adjustment => 'yellow',
            self::Return => 'blue',
            self::SellOut => 'indigo',
            self::SellThrough => 'purple',
        };
    }
}
